client/src/modules/purchases/create/PurchaseOrderForm.js

```diff
@@ -299,10 +302,51 @@ function PurchaseOrderFormService(Inventory, AppCache, Store, Pool, PurchaseOrde
     // configure the PurchaseOrderFormItem with the inventory values
     item.configure(inventoryItem);
 
+    // Inventory item prices are kept in the enterprise currency
+    // Unit prices of items in an order are kept in the selected currency
+    // so we need to fix the unit price.
+    item.unit_price = inventoryItem.price * this.currentExchangeRate;
+
     // make sure to validate and calculate new totals
     this.digest();
   };
 
+  /**
+   * @method setCurrencyId
+   *
+   * @description
+   * Set the currency ID for this order
+   *
+   * @param {number} currencyId
+   */
+  PurchaseOrderForm.prototype.setCurrencyId = function setCurrencyId(currencyId) {
+    this.details.currency_id = currencyId;
+  };
+
+  /**
+   * @method setExchangeRate
+   *
+   * @description
+   * Set the exchange rate for the current currency for all items
+   *
+   * @param {number} newExchangeRate
+   */
+  PurchaseOrderForm.prototype.setExchangeRate = function setExchangeRate(newExchangeRate) {
+    this.currentExchangeRate = newExchangeRate;
+
+    // Update unit prices for all order items
+    this.store.data.forEach((item) => {
+      if (!item._initialised) { return; }
+      // We have no way to tell if this form is being invoked when creating
+      // or editing a purchase, so do this work-around to get the inventory item
+      const avItem = this.inventory.unavailable.get(item.inventory_uuid);
```

Comment:
Hmmm... if this ever comes up again, we may just add a function to the `Pool` that will do this for us.  This is a good solution here.